app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ActionExecutionResult.java

```diff
@@ -38,7 +39,13 @@
     public void setErrorInfo(Throwable error) {
         this.body = error.getMessage();
 
-        if (error instanceof BaseException) {
+        if (error instanceof AppsmithPluginException) {
+            this.statusCode = ((AppsmithPluginException) error).getAppErrorCode().toString();
+            this.title = ((AppsmithPluginException) error).getTitle();
+            this.errorType = ((AppsmithPluginException) error).getErrorType();
+        }
+
+        else if (error instanceof BaseException) {
```

Comment:
Super minor nit: Always have the else clause at the end of the if-block. This ensures that as a reader, I notice the else immediately. If the else block is removed from the if block, it makes code slightly harder to read.
```suggestion
        } else if (error instanceof BaseException) {
```